spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainerTests.java

```diff
@@ -235,6 +248,34 @@ public void publishEvent(ApplicationEvent event) {
 		assertNotNull(failEvent.get());
 		assertThat(failEvent.get(), instanceOf(ListenerContainerConsumerFailedEvent.class));
 		container.stop();
+		cf.destroy();
+	}
+
+	@Test
+	public void testErrorHandler() {
+		brokerRunning.getAdmin().deleteQueue(Q1);
+		Queue q1 = new Queue(Q1, true, false, false, new ArgumentBuilder()
+				.put("x-dead-letter-exchange", "")
+				.put("x-dead-letter-routing-key", DLQ1)
+				.get());
+		brokerRunning.getAdmin().declareQueue(q1);
+		CachingConnectionFactory cf = new CachingConnectionFactory("localhost");
+		DirectMessageListenerContainer container = new DirectMessageListenerContainer(cf);
+		container.setQueueNames(Q1);
+		container.setConsumersPerQueue(2);
+		final AtomicReference<Channel> channel = new AtomicReference<>();
+		container.setMessageListener((ChannelAwareMessageListener) (m, c) -> {
+			channel.set(c);
+			throw new MessageConversionException("intended - should be wrapped in an ARADRE");
+		});
+		container.afterPropertiesSet();
+		container.start();
+		RabbitTemplate template = new RabbitTemplate(cf);
+		template.convertAndSend(Q1, "foo");
+		assertNotNull(template.receive(DLQ1, 10000));
```

Comment:
OK. That took some time for me to figure out that `ConditionalRejectingErrorHandler` is used by default and what it does exactly.
But, eh..., right now it's fine with me.
Pretty smart and cool.

I can merge it with fixes for those minor concerns.
Just let me know!